SourceX/DiabloUI/diabloui.cpp

```diff
@@ -694,91 +720,94 @@ void Render(const UiScrollBar &ui_sb)
 	{
 		const SDL_Rect rect = UpArrowRect(ui_sb);
 		const int frame = static_cast<int>(scrollBarState.upArrowPressed ? ScrollBarArrowFrame_UP_ACTIVE : ScrollBarArrowFrame_UP);
-		DrawArt(rect.x, rect.y, ui_sb.arrow, frame, rect.w);
+		DrawArt(rect.x, rect.y, ui_sb->m_arrow, frame, rect.w);
 	}
 	{
 		const SDL_Rect rect = DownArrowRect(ui_sb);
 		const int frame = static_cast<int>(scrollBarState.downArrowPressed ? ScrollBarArrowFrame_DOWN_ACTIVE : ScrollBarArrowFrame_DOWN);
-		DrawArt(rect.x, rect.y, ui_sb.arrow, frame, rect.w);
+		DrawArt(rect.x, rect.y, ui_sb->m_arrow, frame, rect.w);
 	}
 
 	// Thumb:
-	{
+	if (SelectedItemMax - SelectedItemMin > 0) {
 		const SDL_Rect rect = ThumbRect(
 		    ui_sb, SelectedItem - SelectedItemMin, SelectedItemMax - SelectedItemMin + 1);
-		DrawArt(rect.x, rect.y, ui_sb.thumb);
+		DrawArt(rect.x, rect.y, ui_sb->m_thumb);
 	}
 }
 
-void Render(const UiEdit &ui_edit)
+void Render(const UiEdit *ui_edit)
 {
-	DrawSelector(ui_edit.rect);
-	SDL_Rect rect = ui_edit.rect;
+	DrawSelector(ui_edit->m_rect);
+	SDL_Rect rect = ui_edit->m_rect;
 	rect.x += 43;
 	rect.y += 1;
 	rect.w -= 86;
-	DrawArtStr(ui_edit.value, rect, ui_edit.flags, /*drawTextCursor=*/true);
+	DrawArtStr(ui_edit->m_value, rect, ui_edit->m_iFlags, /*drawTextCursor=*/true);
 }
 
-void RenderItem(UiItem *item)
+void RenderItem(UiItemBase *item)
 {
 	if (item->has_flag(UIS_HIDDEN))
 		return;
-	switch (item->type) {
+	switch (item->m_type) {
 	case UI_TEXT:
-		Render(&item->text);
+		Render((UiText *)item);

```

Comment:
`static_cast<UiText *>` is better because it at least checks that `UiText` is a subclass of `UiItemBase`.